text/010-wasm-pack-watch-and-serve.md

```diff
@@ -0,0 +1,230 @@
+- Start Date: 2019-03-06
+- RFC PR: https://github.com/rustwasm/rfcs/pull/10
+- Tracking Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Add two new sub-commands to `wasm-pack`:
+
+1. `watch`: watch the crate's files and automatically rebuild whenever they
+   change.
+2. `serve`: start an HTTP server for the crate directory on `localhost`.
+
+# Motivation
+[motivation]: #motivation
+
+Enable a smooth and complete local development experience for users who are not
+using a JavaScript bundler.
+
+In particular, we would like to [remove the `npm` and bundler usage from our
+Game of Life tutorial][no-bundler-in-docs] without requiring readers to install
+any additional tools other than the Rust toolchain and `wasm-pack` when setting
+up their development environment. The goal here being less moving parts to learn
+and tools to wrangle when first exploring Rust and WebAssembly.
+
+Additionally, we would like to make sure that the local development server uses
+the `application/wasm` MIME type when serving Wasm binaries. This enables the
+`WebAssembly.instantiateStreaming` fast-path for users.
+
+[no-bundler-in-docs]: https://github.com/rustwasm/book/issues/150
+
+# Stakeholders
+[stakeholders]: #stakeholders
+
+The stakeholders are primarily people that are doing Rust and Wasm development
+without a JavaScript bundler, which potentially will include all of our new
+users who are just starting the Game of Life tutorial. Most of this demographic
+is probably not watching the RFCs repository, and only some are regularly coming
+to working group meetings, so it makes sense to advertise this RFC in TWiRaWA
+and on our @rustwasm Twitter account.
+
+# Detailed Explanation
+[detailed-explanation]: #detailed-explanation
+
+We will add two new subcommands to `wasm-pack`:
+
+1. `watch`
+2. `serve`
+
+## `wasm-pack watch`
+
+The `watch` subcommand will observe the crate directory for changes and
```

Comment:
Perhaps not in v1...? - but I think it would be help to also take a look at the Cargo.toml, look at the dependencie’s and also watch any path dependencies that we see.

Most of my non trivial projects are in a workspace with different path deps - so just watching one crate wouldn’t be of use to me.

I see that there are plans to make use of the build-plan so that would work too. Not familiar enough to know which is an easier approach.

That being said - this seems like a nice step in the right direction - just sharing feedback!